test/e2e/readwrite_test.go

```diff
@@ -0,0 +1,132 @@
+//go:build integration
+// +build integration
+
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/efficientgo/e2e"
+	e2edb "github.com/efficientgo/e2e/db"
+	"github.com/efficientgo/tools/core/pkg/runutil"
+	"github.com/efficientgo/tools/core/pkg/testutil"
+	rulesspec "github.com/observatorium/api/rules"
+)
+
+var sampleRulesA = `
+groups:
+  - name: test-oidc
+    interval: 5s
+    rules:
+      - record: trs
+        expr: vector(1)
+      - alert: HighRequestLatency
+        expr: job:request_latency_seconds:mean5m{job="myjob"} > 0.5
+        for: 10m
+        labels:
+          severity: page
+        annotations:
+          summary: High request latency`
+
+var sampleRulesB = `
+groups:
+  - name: test-oidc
+    interval: 5s
+    rules:
+      - record: btrs
+        expr: vector(1)
+        labels:
+          dummy: yes
+      - alert: HighRequestLatency
+        expr: job:request_latency_seconds:mean5m{job="second"} > 0.5
+        for: 10m`
+
+var invalidRules = `
+groups:
+  - name: test-oidc
+    interval: 5s
+    rules:
+      - record: trs
+        expr: vector(1)
+      - invalid: property`
+
+func TestMetricsReadAndWrite(t *testing.T) {
```

Comment:
Might be a little bit too much copying going on :stuck_out_tongue_winking_eye: Maybe:
```suggestion
func TestRulesReadAndWrite(t *testing.T) {
```